internal/ingress/controller/nginx.go

```diff
@@ -438,8 +438,7 @@ func (n NGINXController) DefaultEndpoint() ingress.Endpoint {
 }
 
 // generateTemplate returns the nginx configuration file content
-func (n NGINXController) generateTemplate(cfg ngx_config.Configuration, ingressCfg ingress.Configuration) ([]byte, error) {
-
+func (n *NGINXController) generateTemplate(cfg *ngx_config.Configuration, ingressCfg *ingress.Configuration) ([]byte, error) {
```

Comment:
I'm not sure I feel comfortable with changing this to a pointer vs a full copy of the structure as it was. While it is a good optimization, generateTemplate shouldn't have write access to the cfg, and the cfg shouldn't be mutated by other processes while being rendered.

Reply:
I revert the params of `generateTemplate` and `OnUpdate`, but I am still trying to use pointers instead of full copies elsewhere.